### ISSUE TITLE
Set start and/or end dates for habits

### ISSUE THREAD

**Issue by holroy** [zoreet/habit-tracker#22]
I'm just thinking if I keep adding habits the list could be long, and if I at some point in time decides to change my habit it would be nice to end the tracking of that habit.

It shouldn't be too hard to implement either, as you could just add a `start-date` and `end-date` into the habit file, and only include if the current date range displayed matches those dates. You wouldn't need to, but you could even mark it somehow in the table, that this habit started/ended at a given date (if that date is within the currently displayed range).

I'm thinking that doing this in combination with me adding the tracker into my daily notes with an end date of the day of the daily note, would be useful as I like to shift my focus every now and then on what I want to track and build a habit towards. If it's not implemented, the display for someone like me (shifting habits to track somewhat often) would fast be cluttered with habits I'm not interested in, or which are so second nature to me that it's not needed to track them anymore).

**Comment by zoreet** [comment]
That's an interesting idea. I will need to think a bit about it, and the best way to approach it.

In the meantime you could solve the problem by moving the "done/archived" habits to another folder. This is how I do it too. I keep my habits organized into folders like this
Habits
--Active
----Habit1.md
----Habit2.md
--Archived -> this is where you keep the ones that you don't want to track anymore
----Habit3.md

And in your daily notes, you can pass the path as "Habits/Active/"

Hope this helps, and thank you for all the comments and PR 🙏

**Comment by holroy** [comment]
The downside of moving them to another folder is that you then loose the history of your habit building, so I'm thinking more in line with creating new habit folders as I move along leaving the old for historic reasons.

If I do that, and keep a little overlap when moving I think it'll look better. But it would be even nicer with start and end dates.

**Comment by YIRU69** [comment]
Agree. Another case, I want to tracker the habit on this week. So I believe it necessary set the start and end dates.

**Comment by zoreet** [comment]
How would you guys expect this to be displayed? After the deadline stop rendering cells? what happens once a habit has no dates to display, do I hide it from the grid in HabitTracker21?

**Comment by holroy** [comment]
If possible consider "greying"/disabling days outside of the range, and completely hide if totally out of the visible ramge of that table.
